src/Agent.Worker/TaskCommandExtension.cs

```diff
@@ -602,6 +614,18 @@ public void Execute(IExecutionContext context, Command command)
                 Boolean.TryParse(isSecretValue, out isSecret);
             }
 
+            String isReadOnlyValue;
+            Boolean isReadOnly = false;
+            if (eventProperties.TryGetValue(TaskSetVariableEventProperties.IsReadOnly, out isReadOnlyValue))
+            {
+                Boolean.TryParse(isReadOnlyValue, out isReadOnly);
+            }
+
+            if (context.TaskVariables.IsReadOnly(name))
+            {
+                context.Warning($"Overwriting readonly variable '{name}'. This behavior will be disabled in the future. See https://github.com/microsoft/azure-pipelines-yaml/blob/master/design/readonly-variables.md for details.");
```

Comment:
same

Reply:
also, should this say 'task variable'? having the exact same error/warning for two different code blocks makes it difficult to triage later on